src/ssd1306-element-element.ts

```diff
@@ -0,0 +1,118 @@
+// Reference: https://cdn-learn.adafruit.com/assets/assets/000/036/494/original/lcds___displays_fabprint.png?1476374574
+import { customElement, html, LitElement, property, SVGTemplateResult } from 'lit-element';
+
+type Dimensions = [number, number];
+
+@customElement('wokwi-ssd1306-element')
+export class Ssd1306ElementElement extends LitElement {
+  @property() dimensions: Dimensions = [141, 108];
+  @property() imageData: ImageData = new ImageData(128, 64);
+
+  putGivenImageData() {
+    const canvas: HTMLCanvasElement | null = document.querySelector('canvas');
+    const ctx = canvas?.getContext('2d');
+    if (!ctx) return;
+    ctx.putImageData(this.imageData, 0, 0);
+  }
+
+  // will be removed!! just for testing
+  testImageData() {
+    const canvas: HTMLCanvasElement | null = document.querySelector('canvas');
+    const ctx = canvas?.getContext('2d');
+    if (!ctx) return;
+    const arr = new Uint8ClampedArray(40000);
+
+    for (let i = 0; i < arr.length; i += 4) {
+      arr[i + 0] = 0; // R value
+      arr[i + 1] = 190; // G value
+      arr[i + 2] = 0; // B value
+      arr[i + 3] = 255; // A value
+    }
+
+    const imageData = new ImageData(arr, 200);
+
+    ctx.putImageData(imageData, 0, 0);
+  }
+
+  firstUpdated() {
+    /// this.putGivenImageData();
+    // Getting null for canvas, there's an issue here
+    this.testImageData();
```

Comment:
@urish maybe you'll have an idea, at this point I expected to have the rendered DOM with the canvas element, but querying it will resolve as null so I can't test the image data, (+ the size of the foreignObject and hosted canvas isn't right as well)

https://codepen.io/lironhazan/pen/VwLJgYK